spec.html

```diff
@@ -18,6 +18,110 @@ <h1>Array Objects</h1>
+                    1. <del>Let _xSmaller_ be ! IsLessThan(_xString_, _yString_, *true*).</del>
+                    1. <del>If _xSmaller_ is *true*, return *-1*<sub>𝔽</sub>.</del>
+                    1. <del>Let _ySmaller_ be ! IsLessThan(_yString_, _xString_, *true*).</del>
+                    1. <del>If _ySmaller_ is *true*, return *1*<sub>𝔽</sub>.</del>
+                    1. <del>Return *+0*<sub>𝔽</sub>.</del>
+                    1. <ins>Return ? CompareArrayElements(_x_, _y_, _comparefn_).</ins>
+                1. <del>Return ? SortIndexedProperties(_obj_, _len_, _SortCompare_).</del>
+                1. <ins>Let _sortedList_ be ? SortIndexedProperties(_obj_, _len_, _SortCompare_, *true*).</ins>
+                1. <ins>Let _itemCount_ be the number of elements in _sortedList_.</ins>
+                1. <ins>Let _j_ be 0.</ins>
+                1. <ins>Repeat, while _j_ &lt; _itemCount_,</ins>
+                  1. <ins>Perform ? CreateDataPropertyOrThrow(_obj_, ! ToString(𝔽(_j_)), _sortedList_[_j_]).</ins>
+                  1. <ins>Set _j_ to _j_ + 1.</ins>
+                1. <ins>Repeat, while _j_ &lt; _len_,</ins>
+                  1. <ins>Perform ? DeletePropertyOrThrow(_obj_, ! ToString(𝔽(_j_))).</ins>
+                  1. <ins>Set _j_ to _j_ + 1.</ins>
+                1. <ins>Return _obj_.</ins>
+                </emu-alg>
+            </emu-clause>
+
+            <emu-clause id="sec-comparearrayelements" type="abstract operation">
+                <h1>
+                  CompareArrayElements(
+                    _x_: unknown,
+                    _y_: unknown,
+                    _comparefn_: unknown
+                  ): either a normal completion containing a Number or an abrupt completion
+                </h1>
+                <dl class="header">
+                </dl>
+                <emu-alg>
+                    1. If _x_ and _y_ are both *undefined*, return *+0*<sub>𝔽</sub>.
+                    1. If _x_ is *undefined*, return *1*<sub>𝔽</sub>.
+                    1. If _y_ is *undefined*, return *-1*<sub>𝔽</sub>.
+                    1. If _comparefn_ is not *undefined*, then
+                        1. Let _v_ be ? ToNumber(? Call(_comparefn_, *undefined*, &laquo; _x_, _y_ &raquo;)).
+                        1. If _v_ is *NaN*, return *+0*<sub>𝔽</sub>.
+                        1. Return _v_.
+                    1. [id="step-sortcompare-tostring-x"] Let _xString_ be ? ToString(_x_).
+                    1. [id="step-sortcompare-tostring-y"] Let _yString_ be ? ToString(_y_).
+                    1. Let _xSmaller_ be ! IsLessThan(_xString_, _yString_, *true*).
+                    1. If _xSmaller_ is *true*, return *-1*<sub>𝔽</sub>.
+                    1. Let _ySmaller_ be ! IsLessThan(_yString_, _xString_, *true*).
+                    1. If _ySmaller_ is *true*, return *1*<sub>𝔽</sub>.
+                    1. Return *+0*<sub>𝔽</sub>.
```

Comment:
any reason not to use `localeCompare` logic here? (does the default sort do this, or does it use `<`?)

Reply:
I took this from the current spec for `Array.prototype.sort`, should be direct copy pasta.